### PR TITLE
Adds the delete palette confirmation modal and functionality

### LINKED ISSUE #23: Optional Extension (Delete Confirmation)
When a user tries to delete a saved palette, have them confirm that they really do want to delete it.

### PR DESCRIPTION
**Pull request by mbenfowler**
## What is the feature?
This adds a user modal asking for confirmation when attempting to delete a saved palette.

## How you implemented the solution?
The meat of the changes here is `await`-ing for a `Promise` from the user interaction within the confirmation modal.
I encourage you to read up on this new tech, since it was the only way I could find to pause within an already firing eventListener, and wait for an interaction from another listener.
[https://javascript.info/asyn](https://javascript.info/async)

## Any changes in the UI (Screenshots)?
Screenshot 2023-04-14 at 3 46 23 PM

## How to test the feature (A test scenario or any new setup)?
- Hitting the `X` in the top right corner of the modal will not delete the palette and exit the modal.
- Hitting the `Delete this palette` button confirms deletion of the palette and exits the modal.
  - Note that the palette deleted no longer appears in the 
- When the modal is open, interactions with other sections of the app should not be allowed!

### CODE REVIEW

**Review by jalbe0076 — CHANGES_REQUESTED** [review]
After dealing with the merge conflict we'll have to fix a few things
We'll need to add some functionality to block event listeners so that we can't just ignore the pop up. When the confirmation window is open we can still save palettes, get new palettes, lock buttons etc. 
image

If you click any X on the screen the pop up will disappear. 

This is subjective, but I don't like the location of the pop up, this can be discussed.

File: main.js
Comment (jalbe0076) [inline main.js:48]: There are a lot of `.toggle` methods, we should consider this to use a function when refactoring

File: styles.css
Comment (jalbe0076) [inline styles.css]: We only need translate: -50% -50% because we don't need the z axis

File: main.js
Comment (jalbe0076) [inline main.js]: refactor to not use arrow function `function listener(event)`

**Review by jalbe0076 — CHANGES_REQUESTED** [review]

**Review by jalbe0076 — APPROVED** [review]
Looks great!!! Nice job on the position and making the rest blocked until we get the user response

**Review by SethBurns — APPROVED** [review]
Spoke about this online. Well done man! Cool stuff

**Pull request merged**
